lib/ilp.js

```diff
@@ -96,10 +102,26 @@ class Ilp {
 
     this.log.info('Connection created successfully')
 
-    await this.sendMoney({ ilpAmount: ilpScaledAmount, connection })
+    let { success, remainingAmount, error } = await this.sendMoney({ ilpAmount: ilpScaledAmount, connection })
+
+    // If we have sent NO money and the ilp sender failed, throw an error so we're notified of the failure
+    if (error && remainingAmount === ilpScaledAmount) {
+      throw new Error('ERROR: Encountered error when sending money via ILP: ', error)
+    }
+
+    // Simply log error, still continue noting the remaining amount but just log error for debugging purposes
+    if (error) {
+      this.log.info('ERROR: Encountered error when sending money via ILP: ', error)
+    }
+
+    // Reset remaining amount to millicents, where as ILP operates in nanodollars
+    if (remainingAmount > 0) remainingAmount = remainingAmount / 10000
 
     this.log.info('Closing connection to ILP connector...')
     await connection.end()
+
+    // Pass along success and remaining amount to caller so they can handle it
+    return { success, remainingAmount: remainingAmount || 0 }
```

Comment:
what's the thought behind `remainingAmount || 0` ?

Reply:
in case it's undefined i want it to be guaranteed a number